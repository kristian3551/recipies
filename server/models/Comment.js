const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const Model = mongoose.model;
const { String, Number } = Schema.Types;

const commentSchema = new Schema({
    content: {
        type: String
    },
    author: {
        type: String
    },
    createdAt: {
        type: Date,
        default: Date.now()
    }
});

module.exports = new Model('Comment', commentSchema);