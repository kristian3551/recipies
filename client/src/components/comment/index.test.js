import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Comment from './index';

describe('<Comment/>', function () {
    it('should render right things', function () {
        const component = renderer.create(<Comment comment={
            {
                createdAt: '01 Jan 1970 00:00:00 GMT',
                author: 'kristian01',
                content: 'asdasdasd'
            }
        } />);
        expect(component.toJSON()).toMatchSnapshot();
    });
})