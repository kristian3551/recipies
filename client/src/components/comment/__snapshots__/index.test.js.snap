// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment/> should render right things 1`] = `
<div
  className="new_comment"
>
  <ul
    className="user_comment"
  >
    <div
      className="comment_body"
    >
      <p>
        asdasdasd
      </p>
    </div>
    <div
      className="comment_toolbar"
    >
      <div
        className="comment_details"
      >
        <ul>
          <li
            style={
              Object {
                "color": "black",
              }
            }
          >
            <i
              className="fa fa-clock-o"
              style={
                Object {
                  "color": "black",
                }
              }
            />
             
            3
            :
            0
          </li>
          <li
            style={
              Object {
                "color": "black",
              }
            }
          >
            <i
              className="fa fa-calendar"
            />
            1
            /
            1
            /
            1970
          </li>
          <li
            style={
              Object {
                "color": "black",
              }
            }
          >
            <i
              className="fa fa-pencil"
            />
             
            <span
              className="user"
            >
              kristian01
            </span>
          </li>
        </ul>
      </div>
    </div>
  </ul>
</div>
`;
